### PR TITLE
Implement a sanity check module for DocGenerator

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#1628|

## Description
This PR adds the new module Sanity to **DocGenerator** tool.
At the moment, this module is executed independently from the main DocGenerator tool with `python Sanity.py`
This module checks if every Mandatory field is present in the documentation jsons, identifies the different tags found and counts every test and test module to obtain a coverage calculus.

### CODE REVIEW

File: docs/DocGenerator/Config.py [hunk 1]
```diff
@@ -14,10 +14,14 @@ def __init__(self):
...
-        self.required_fields = self.__fields()
-        self.ignored_fields = self.__fields()
-        with open(CONFIG_PATH) as fd:
-            self.__config_data = yaml.load(fd)
+        self.module_fields = self.__fields()
+        self.test_fields = self.__fields()
+
+        try:
+            with open(CONFIG_PATH) as fd:
+                self.__config_data = yaml.load(fd)
```
Comment (MiguelazoDS) [inline docs/DocGenerator/Config.py:22]: We changed this to safe_load() because load() is deprecated.

File: docs/DocGenerator/Sanity.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                    with open(os.path.join(root,test_file)) as fd:
+                        file_content = fd.read()
+                    module = ast.parse(file_content)
+                    functions = [node for node in module.body if isinstance(node, ast.FunctionDef)]
+                    for function in functions:
+                        for regex in function_regexes:
+                            if regex.match(function.name):
+                                self.project_tests = self.project_tests + 1
+
+
```
Comment (MiguelazoDS) [inline docs/DocGenerator/Sanity.py:75]: Too many spaces.

File: docs/DocGenerator/Config.py (hunk 1 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/Config.py]: Fixed!

File: docs/DocGenerator/Sanity.py (hunk 2 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/Sanity.py]: Fixed

**Review by MiguelazoDS — APPROVED** [review]

**Pull request merged**